app/controllers/badges_controller.rb

```diff
@@ -4,6 +4,9 @@ class BadgesController < ApplicationController
 
   def show
     @badge = Badge.find_by_slug(params[:slug])
+    if @badge.nil?
+      render file: "#{Rails.root}/public/404.html", layout: false, status: :not_found
```

Comment:
I think you can just call https://github.com/thepracticaldev/dev.to/blob/master/app/controllers/application_controller.rb#L13 

You can see the pattern here:

https://github.com/thepracticaldev/dev.to/blob/f93af16fbe99a83d5fd3e74bd7ffa3a2532ae6b4/app/controllers/chat_channels_controller.rb#L19-L23

Reply:
Cool, thanks for the pointer!

Reply:
I was thrown off a little by the `RoutingError`, but looks like production env will rescue it and render 404 right?